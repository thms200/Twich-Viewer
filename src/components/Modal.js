import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import styles from '../containers/AppContainer.module.css';
const cx = classNames.bind(styles);

export default function Modal({ activeFlag, children }) {
  return (
    <div className={cx({ nonVisible: !activeFlag, visible: activeFlag })}>
      {children}
    </div>
  );
}

Modal.propTypes = {
  activeFlag: PropTypes.bool.isRequired,
  children: PropTypes.object.isRequired,
};
