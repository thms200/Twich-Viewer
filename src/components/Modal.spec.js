import React from 'react';
import renderer from 'react-test-renderer';
import Modal from './Modal';

describe('<Modal />', () => {
  const nonActiveModal = renderer
    .create(
      <Modal
        activeFlag={false}
      >
        {<div>modal children</div>}
      </Modal>
    )
    .toJSON();

  it('renders correctly', () => {
    expect(nonActiveModal).toMatchSnapshot();
  });
});
