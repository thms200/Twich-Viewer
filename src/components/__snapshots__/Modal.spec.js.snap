// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> renders correctly 1`] = `
<div
  className="nonVisible"
>
  <div>
    modal children
  </div>
</div>
`;
